lib/auth/auth_chalresp.c

```diff
@@ -100,32 +101,67 @@ struct auth_chalresp_result {
 #pragma pack(pop)
 
 
+
 /**
- * Shared key.
- * @brief  In a production system, the shared key should be stored in a
- * secure hardware store such as an ECC608A or TrustZone.
+ * Challenge-Response instance info.  Used to set shared key for each
+ * auth instance.
  */
-static uint8_t default_shared_key[AUTH_SHARED_KEY_LEN] = {
-	0xBD, 0x84, 0xDC, 0x6E, 0x5C, 0x77, 0x41, 0x58, 0xE8, 0xFB, 0x1D, 0xB9, 0x95, 0x39, 0x20, 0xE4,
-	0xC5, 0x03, 0x69, 0x9D, 0xBC, 0x53, 0x08, 0x20, 0x1E, 0xF4, 0x72, 0x8E, 0x90, 0x56, 0x49, 0xA8
+struct chalresp_instance {
+	bool in_use;
+	uint8_t chalresp_key[AUTH_SHARED_KEY_LEN];
 };
 
-/* default shared key if not set */
-static uint8_t *shared_key = default_shared_key;
+static struct chalresp_instance chalresp_inst_tbl[NUM_CHALLENGE_RESP_INST];
```

Comment:
Found problem here were the Challenge-Response didn't correctly handle multiple instances.  Now it does.